src/builtins/core/date.rs

```diff
@@ -82,8 +82,62 @@ impl PartialDate {
     crate::impl_with_fallback_method!(with_fallback_year_month, () PlainYearMonth); // excludes day
     crate::impl_with_fallback_method!(with_fallback_date, (with_day: day) PlainDate);
     crate::impl_with_fallback_method!(with_fallback_datetime, (with_day:day) PlainDateTime);
+    crate::impl_with_fallback_method_zoned!(with_fallback_zoneddatetime, (with_day:day), ZonedDateTime);
+}
+
+#[doc(hidden)]
+#[macro_export]
+macro_rules! impl_with_fallback_method_zoned {
```

Comment:
suggestion: change this to a normal method over a macro.

Since this is a one off and can't really take advantage of the other macro, it would probably be better for this to be a direct method implementation on `PartialDate` so that it's easy to read and reason about in the future.